### PR TITLE
lilypond: fix $PATH

### PR DESCRIPTION
**Pull request by astro**
## Description of changes

Fix for:
```
$ musicxml2ly T.mxl
.musicxml2ly-wrapped: Reading MusicXML from T.mxl ...
.musicxml2ly-wrapped: Input file T.mxl is compressed, extracting raw MusicXML data
.musicxml2ly-wrapped: Converting to LilyPond expressions...
.musicxml2ly-wrapped: Converting to LilyPond expressions...
.musicxml2ly-wrapped: Converting to LilyPond expressions...
.musicxml2ly-wrapped: Converting to LilyPond expressions...
.musicxml2ly-wrapped: Converting to LilyPond expressions...
.musicxml2ly-wrapped: Converting to LilyPond expressions...
.musicxml2ly-wrapped: Output to `T.ly'
/nix/store/6payx2da66dbjl6vg15csxfb5hpf3df4-bash-5.2-p15/bin/sh: line 1: cut: command not found
/nix/store/6payx2da66dbjl6vg15csxfb5hpf3df4-bash-5.2-p15/bin/sh: line 1: lilypond: command not found
/nix/store/6payx2da66dbjl6vg15csxfb5hpf3df4-bash-5.2-p15/bin/sh: line 1: head: command not found
.musicxml2ly-wrapped: Converting to current version () notations ...
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/misc/lilypond/default.nix [hunk 1]
```diff
@@ -20,7 +20,7 @@ stdenv.mkDerivation rec {
         # Override default argv[0] setting so LilyPond can find
         # its Scheme libraries.
         wrapProgram "$f" --set GUILE_AUTO_COMPILE 0 \
-                         --set PATH "${ghostscript}/bin" \
+                         --set PATH "${ghostscript}/bin:${coreutils}/bin:$out/bin" \
```
Comment (yurrriq) [inline pkgs/misc/lilypond/default.nix:23]: Is this needed for all of `$out/bin/*` or only some, such as `musicxml2ly`? I guess I only (directly) use `lilypond` in practice and haven't hit this issue.

Comment (astro) [reply pkgs/misc/lilypond/default.nix:23]: Yeah, only for some of the entries.

However, I am not in favor of creating an additional maintenance burden by differentiating between them.

Comment (SuperSandro2000) [inline pkgs/misc/lilypond/default.nix:23]:
```suggestion
        wrapProgram "$f" \
          --set GUILE_AUTO_COMPILE 0 \
          --set PATH "${lib.makeBinPath [ ghostscript coreutils ]" \
```

Comment (astro) [reply pkgs/misc/lilypond/default.nix]: You forgot `$out`, I added it

**Review by yurrriq — APPROVED** [review]
lgtm now. what do you think, @SuperSandro2000?

File: pkgs/misc/lilypond/default.nix [hunk 2]
```diff
@@ -19,9 +19,10 @@ stdenv.mkDerivation rec {
     for f in "$out/bin/"*; do
         # Override default argv[0] setting so LilyPond can find
         # its Scheme libraries.
-        wrapProgram "$f" --set GUILE_AUTO_COMPILE 0 \
-                         --set PATH "${ghostscript}/bin" \
-                         --argv0 "$f"
+        wrapProgram "$f" \
+          --set GUILE_AUTO_COMPILE 0 \
+          --set PATH "${lib.makeBinPath [ ghostscript coreutils "$out" ]}" \
```
Comment (Aleksanaa) [inline pkgs/misc/lilypond/default.nix:24]:
```suggestion
          --set PATH "${lib.makeBinPath [ ghostscript coreutils (placeholder "out") ]}" \
```

Maybe this is better? Dealing with bash variables in Nix functions is dangerous.

Comment (SuperSandro2000) [reply pkgs/misc/lilypond/default.nix:24]: > Dealing with bash variables in Nix functions is dangerous.

We are doing this all over the place.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**